src/pytest_env/plugin.py

```diff
@@ -20,19 +29,31 @@ def pytest_load_initial_conftests(
     """Load environment variables from configuration files."""
     for line in early_config.getini("env"):
         part = line.partition("=")
-        key = part[0].strip()
+        # INI key consists of flags and of the env variable key
+        # For example D:R:NAME=VAL has two flags (R and D), NAME key, and VAL value
+        ini_key = part[0].strip()
         value = part[2].strip()
 
-        # Replace environment variables in value. for instance  TEST_DIR={USER}/repo_test_dir.
-        value = value.format(**os.environ)
+        ini_key_parts = ini_key.split(":")
+        flags = ini_key_parts[:-1]
+
+        invalid_flags = set(flags) - _ALLOWED_FLAGS
+        if invalid_flags:
+            raise Exception(f"Unrecognized flags {','.join(sorted(invalid_flags))}")
```

Comment:
This has no test. Also I'm not sure we should do this, would be backwards incompatible change and people might have alpha:two=X lines today that suddenly becomes invalid. No? Only d and r are special we should let rest be?

Reply:
Do you mean that people might have colons (`:`) in their variable names? A lot of places (like bash, kubernetes) don't allow colons in keys at all.

> Only d and r are special we should let rest be

But still, I agree

Reply:
Be that so, but I think in environment variables : makes sense to allow. Another angle to consider is that I'm thinking about freezing pytest.ini as is, and create a new configuration for pyproject.toml that doesn't require such heuristics. It supports natively dictionaries so we don't need the R, and for defaults we can make dictionary values be a list where the second value is the default.